source_ds/mm_main_ds.c

```diff
@@ -19,10 +19,183 @@
  *
  ******************************************************************************/
 
+#include "maxmod.h"
 #include "mm_main.h"
+#include "mp_mas_structs.h"
 #include "mm_main_ds.h"
+#include "mm_comms.h"
+#include "mm_comms7.h"
+#include "mm_mas.h"
+#include "mm_effect.h"
+#include "mp_defs.h"
+#include "libnds_defs.h"
+
+#if defined(SYS_GBA)
+#include "mm_main_gba.h"
+#include "mm_mixer_gba.h"
+#elif defined(SYS_NDS)
+#include "mm_main_ds.h"
+#include "mm_mixer_ds.h"
+#include "mm_mixer_super.h"
+#endif
 
 void mmInit7(void);
+void mmInitialize(mm_bool);
+mm_word mmEventForwarder(mm_word, mm_word);
+void StopActiveChannel(mm_word);
+
+#define NUM_CHANNELS 32
+
+#define INITIALIZED_VALUE 42
+#define BASE_VOLUME 0x400
+#define BASE_TEMPO 0x400
+#define BASE_PITCH 0x400
+#define ALL_CHANNELS_UNLOCK 0xFFFF
+
+#define TIMER_0_IRQ (1<<3)
+
+extern mm_byte mm_mixing_mode;
+
+mm_module_channel mm_rds_pchannels[NUM_CHANNELS];
+mm_active_channel mm_rds_achannels[NUM_CHANNELS];
+
+// Returns true if the system is ready for playback
+mm_bool mmIsInitialized(void) {
+    return mmInitialized == INITIALIZED_VALUE;
+}
+
+void mmInitialize(mm_bool do_initialize) {
```

Comment:
Tbh, it's to have it all in the same place, instead of using the value in different places thoughout the code. Making this an inline should fix that with no problem.

Reply:
Does maxmod really write to this variable in other places beyond `mmInit`? Should it be doing so?

Reply:
Just in mmInstall. But it does use it twice in mmInit (don't ask me why, I'm wondering it myself).